### ISSUE TITLE
reposync with kickstart does not seem to work.

### ISSUE THREAD

**Issue by sbluhm** [uyuni-project/uyuni#3134]
# Additional Information

Running `reposync` on `http://msync.centos.org/centos/8/BaseOS/x86_64/os/` gives me this error:

```
# /usr/bin/spacewalk-repo-sync --channel centos8-baseos-x86_64 --type yum --non-interactive --sync-kickstart
08:09:43 ======================================
08:09:43 | Channel: centos8-baseos-x86_64
08:09:43 ======================================
08:09:43 Sync of channel started.
Metadaten von Repository 'centos8-baseos-x86_64' abrufen ..................................................................................................................................................[fertig]
Cache für Repository 'centos8-baseos-x86_64' erzeugen .....................................................................................................................................................[fertig]
Alle Repositorys wurden aktualisiert.
08:09:46 Repo URL: http://msync.centos.org/centos/8/BaseOS/x86_64/os/
08:09:46     Packages in repo:              1697
08:09:51     No new packages to sync.
08:09:51
08:09:51   Patches in repo: 0.
08:09:52
08:09:52   Importing kickstarts.
08:09:52 Trying treeinfo
08:09:52 Unexpected error: <class 'AttributeError'>
08:09:52 Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/repo_plugins/yum_src.py", line 1175, in get_file
    downloaded = urlgrabber.urlgrab(path, temp_file)
  File "/usr/lib/python3.6/site-packages/urlgrabber/grabber.py", line 787, in urlgrab
    return default_grabber.urlgrab(url, filename, **kwargs)
  File "/usr/lib/python3.6/site-packages/urlgrabber/grabber.py", line 1203, in urlgrab
    raise err
urlgrabber.grabber.URLGrabError: [Errno 2] Local file does not exist: /root/treeinfo

```
Checking the variables `path` and `temp_file` variables give me this (just before calling `urlgrabber.urlgrab`):
```
path=treeinfo
temp_file=/var/cache/rhn/reposync/1/centos8-baseos-x86_64/treeinfo..download
```

## Version of Uyuni Server and Proxy (if used)
`zypper info Uyuni-Server-release`
```
Repository         : uyuni-server-devel
Name               : Uyuni-Server-release
Version            : 2020.11-104.5.uyuni
Arch               : x86_64
Anbieter           : obs://build.opensuse.org/systemsmanagement:Uyuni

```

## Details about issue
Setup of the repo and channel were done via the Web UI.

**Comment by sbluhm** [comment]
I don't think this worked with Uyuni since Jan 2019.

Original Spacewalk code used `yum.yumRepo.YumRepository(name).grab.urlgrab(path, temp_file)` instead of https://github.com/uyuni-project/uyuni/blob/10f41e7e8eeff5db8ae46db65ec74beab6ec88fb/backend/satellite_tools/repo_plugins/yum_src.py#L1176 which just downloads a file.

The `yum` `urlgrab` function has more logic to get source URLs.
https://github.com/rpm-software-management/yum/blob/f8616a2d6e22705371fe6ba47597238d3d1dc2f1/yum/yumRepo.py#L706
https://github.com/rpm-software-management/yum/blob/f8616a2d6e22705371fe6ba47597238d3d1dc2f1/yum/yumRepo.py#L699
https://github.com/rpm-software-management/yum/blob/f8616a2d6e22705371fe6ba47597238d3d1dc2f1/yum/yumRepo.py#L608

Let me see if we can get a kickstart specific feature into Uyuni.

**Comment by sbluhm** [comment]
@mcalmer , I just grepped through the repository and as far as I understand, the function
https://github.com/uyuni-project/uyuni/blob/fed9c22db65c077e8aa5a98cb075451b6db1a217/backend/satellite_tools/repo_plugins/yum_src.py#L1164
is only used in [backend/satellite_tools/reposync.py](https://github.com/uyuni-project/uyuni/blob/master/backend/satellite_tools/reposync.py) and only in conjunction with kickstart. Can you please confirm that?
I am asking to keep risks of braking other code to a minimum (and am still wondering why the change if it never worked)...

**Comment by mcalmer** [comment]
@sbluhm Yes, it seems you are right. Only used in reposync.py and only in conjunction with kickstart.

**Comment by sbluhm** [comment]
Fixed in #3149 . Maybe not as robust as the original yum handled mirror lists. Now only the repo url is added.
Not sure if this causes any issues. Then the logic should be more complex. But I doubt anyone but me is using this feature or someone would have reported this issue before.
